include/circt/Dialect/FIRRTL/FIRRTLDeclarations.td

```diff
@@ -316,7 +316,8 @@ def RegOp : ReferableDeclOp<"reg", [Forceable]> {
       $_state.addTypes(elementType);
       if (forceable) {
         $_state.addAttribute(getForceableAttrName($_state.name), forceable);
-        $_state.addTypes(detail::getForceableResultType(true, elementType));
+        if (auto type = detail::getForceableResultType(true, elementType))
```

Comment:
Don't like it, but otherwise we may end up creating a TypeRange w/null in it which causes assertion failure.

Instead, let verifier reject.